lexers/todotxt.lua

```diff
@@ -0,0 +1,56 @@
+-- Copyright 2025 Chris Clark
+-- todo.txt https://github.com/too-much-todotxt/spec LPeg lexer.
+
+local lexer = lexer
+local P, S = lpeg.P, lpeg.S
+
+local lex = lexer.new('todotxt')
+
+local not_whitespace = lexer.any - lexer.space - P(':')
+local not_whitespace_word = not_whitespace * not_whitespace^0
+
+
+-- Done/Complete items, map to comment style
+lex:add_rule('done', lex:tag(lexer.COMMENT, lexer.starts_line(lexer.to_eol('x '))))
+
+-- Priority, treat A, B, C as unique, D+ same style
+lex:add_rule('priority_A', lex:tag(lexer.ERROR, lexer.starts_line('(A) ')))
+lex:add_rule('priority_B', lex:tag(lexer.PREPROCESSOR, lexer.starts_line('(B) ')))
+lex:add_rule('priority_C', lex:tag(lexer.NUMBER, lexer.starts_line('(C) ')))
+lex:add_rule('priority', lex:tag(lexer.BOLD, lexer.starts_line(P('(') * lexer.upper * P(') '))))
+
+-- Idea, lump some priority styles together
+--lex:add_rule('priority', lex:tag(lexer.NUMBER, lexer.starts_line('(A) ') + lexer.starts_line('(B) ') + lexer.starts_line('(C) ') ))
+
+
+-- key:value
+-- https://github.com/too-much-todotxt/spec/issues/23
+-- TODO different style for key and value so they are clearly marked?
```

Comment:
This can be an opportunity for something like:

```
local key = lex:tag(lexer.KEYWORD, not_whitespace_word)
local colon = lex:tag(lexer.OPERATOR, P(':'))
local value = lex:tag(lexer.CONSTANT, not_whitespace_word)
lex:add_rule('key_value', key * colon * value)
```

Reply:
I've taken this as-is (with some doc comments with two other options), thanks! This triggered a review of the date style.